GTFO-API/Utilities/LiveEdit.cs

```diff
@@ -0,0 +1,143 @@
+        /// <param name="includeSubDir">Include Sub-directories?</param>
+        public static LiveEditListener CreateListener(string path, string filter, bool includeSubDir)
+        {
+            var listener = new LiveEditListener(path, filter, includeSubDir);
+            m_Listeners.Add(listener);
+            return listener;
+        }
+
+        /// <summary>
+        /// Deallocate the LiveEdit Listener
+        /// </summary>
+        /// <param name="listener">Listener to deallocate</param>
+        public static void DeallocateListener(LiveEditListener listener)
+        {
+            if (listener == null)
+            {
+                APILogger.Error("LiveEdit", "DeallocateListener - listener was null!");
+                return;
+            }
+
+            if (listener.m_Watcher == null)
+            {
+                APILogger.Error("LiveEdit", "DeallocateListener - listener was already deallocated!");
+                return;
+            }
+
+            m_Listeners.Remove(listener);
+            listener.Deallocate();
+        }
+    }
+
+    /// <summary>
+    /// 
+    /// </summary>
+    [SuppressMessage("Interoperability", "CA1416:Platform Compatible on FileSystemWatcher", Justification = "GTFO is Windows only game")]
+    public sealed class LiveEditListener
+    {
+        /// <summary>
+        /// Event when File has Changed
+        /// </summary>
+        public event LiveEditEventHandler FileChanged;
+        /// <summary>
+        /// Event when File has Deleted
+        /// </summary>
+        public event LiveEditEventHandler FileDeleted;
+        /// <summary>
+        /// Event when File has Created
+        /// </summary>
+        public event LiveEditEventHandler FileCreated;
+        /// <summary>
+        /// Event when File has Renamed
+        /// </summary>
+        public event LiveEditEventHandler FileRenamed;
+
+        internal FileSystemWatcher m_Watcher = null;
+
+        private LiveEditListener() { }
+
+        internal LiveEditListener(string path, string filter, bool includeSubDir)
+        {
+            m_Watcher = new ()
+            {
+                Path = path,
+                Filter = filter,
+                IncludeSubdirectories = includeSubDir,
+                NotifyFilter = NotifyFilters.FileName | NotifyFilters.LastWrite | NotifyFilters.CreationTime
+            };
+
+            m_Watcher.Changed += (sender, e) => { FileChanged?.Invoke(e); };
+            m_Watcher.Deleted += (sender, e) => { FileDeleted?.Invoke(e); };
+            m_Watcher.Created += (sender, e) => { FileCreated?.Invoke(e); };
+            m_Watcher.Renamed += (sender, e) => { FileRenamed?.Invoke(e); };
+            m_Watcher.Error += (sender, e) =>
+            {
+                APILogger.Error("LiveEdit", $"Path: {path} error reported! - {e.GetException()}");
+            };
+
+            StartListen();
+        }
+
+        internal void Deallocate()
```

Comment:
This could also do the `s_Listeners.Remove(..)` call with `this`. Would prevent accidental disposal without removal from list and it would remove unnecessary public api